src/anyio/_core/_synchronization.py

```diff
@@ -792,41 +813,72 @@ def from_max_per_second(cls, max_per_second: int, /) -> Self:
         """
         return cls(1, 1 / max_per_second)
 
+    def _update_available(self, slot_time: float | None = None) -> bool:
```

Comment:
I think this method should only perform the availability computation logic and return the new available slots and next time frame, and not change the internal state. Only `acquire()` should change the state.